ephios/extra/permissions.py

```diff
@@ -1,10 +1,27 @@
+from functools import wraps
+
 from django.contrib.auth.models import Group, Permission
+from django.core.exceptions import PermissionDenied
 from django.forms import BooleanField
 from guardian.ctypes import get_content_type
 from guardian.shortcuts import assign_perm, remove_perm
 from guardian.utils import get_group_obj_perms_model
 
 
+def staff_required(view_func):
+    @wraps(view_func)
+    def _wrapped_view(request, *args, **kwargs):
+        if request.user.is_staff:
+            return view_func(request, *args, **kwargs)
+        if not request.user.is_authenticated:
+            from django.contrib.auth.views import redirect_to_login
+
+            return redirect_to_login(request.get_full_path())
+        raise PermissionDenied
+
+    return _wrapped_view
```

Comment:
I gather that `staff_member_required` redirects to the admin login and isn't the right choice here then?
Did you copy this code from somewhere?

Reply:
exactly.
nope, self-written but I looked at user_passes_test to figure out how a decorator works ^^